debian/control

```diff
@@ -0,0 +1,27 @@
+Source: nymea-plugin-ws281xspi
+Section: utils
+Priority: options
+Maintainer: Hermann Detz <hermann.detz@gmail.com>
+Build-depends: debhelper (>= 9.0.0),
+               libnymea1-dev,
+               nymea-dev-tools:native,
+               qt5-qmake:native,
+               pkg-config,
+               scons,
```

Comment:
This fails to install when cross-building. Afaict it should be `scons:native` instead?

Reply:
Hmm... what does scons actually do?

Reply:
scons is the build system of the library that is packaged in the submodule. My initial idea was to keep their build system to avoid side effects if they e.g. split up functionality into different files or something similar. If it causes troubles regarding dependencies, we can as well integrate it into qmake.

Reply:
Since the container for crossbuilder is now working, I'm back to this problem. I changed scons to scons:native, which results in unmet dependencies. Couldn't find too much information on using scons within the deb package build system - actually it's discouraged.

```
Using DEB_BUILD_PROFILES: nocheck nodoc noautodbgsym
Building nymea-plugin-ws281xspi for armhf and deploying to device.
LXD container nymea-plugin-ws281xspi-stretch-amd64-armhf already exists.
Error: Common start logic: The container is already running

armhf build dependencies for nymea-plugin-ws281xspi already installed in container nymea-plugin-ws281xspi-stretch-Building nymea-plugin-ws281xspi for armhf in parallel (-j5).
dpkg-buildpackage: info: source package nymea-plugin-ws281xspi
dpkg-buildpackage: info: source version 0.0.1+202010071123~a7d95d7~stretch
dpkg-buildpackage: info: source distribution UNRELEASED
dpkg-buildpackage: info: source changed by  <hermann@base-guh-builder-stretch-amd64-armhf>
dpkg-buildpackage: info: host architecture armhf
 dpkg-source -I -Iobj-* -Idebian/tmp/* -I.bzr* --before-build nymea-plugin-ws281xspi
dpkg-checkbuilddeps: error: Unmet build dependencies: scons:native
dpkg-buildpackage: warning: build dependencies/conflicts unsatisfied; aborting
dpkg-buildpackage: warning: (Use -d flag to override.)

```

So the next best guess would be to include the files into the plugin's project file. Yet this would rely on the assumption that the file structure does not change in future versions.

Or as another option to play it safe, include the source code files of the library in their current version into the repo here.

Reply:
Well, when using a git submodule, the used commit hash is stored along with it. So any checkout would always use the same revision. Unless the project devs force-push to their master branch and break the history, adding the files to the .pro in here seems to be fine IMO. If you see the risk of them breaking their master branch history, we might as well fork it on github and use our fork as the submodule.